### DISCUSSION TITLE
Hot Water Tank thoughts

### DISCUSSION THREAD

**Discussion by r200cjt** [ExperienceLovelace/ha-floorplan#359] [Q&A]
Hey guys, long time user - first time poster. I've an idea of what I want to do - I'm just really unsure how to go about doing it. I set up a floorplan for my home years ago, used it every day - took a bit of a break from automating the house, just enjoyed living with it! But now I've got the bug again and I'm looking build upon what I have.

I decided I wanted more information on my hot water tank, I installed some sensors and wanted to depict that information with a visual of my tank. I have copied some code and generated SVG images running off an apache server on my network, I just really dislike how it runs - it reloads every 20 seconds and makes my dashboard look unsightly when it does. So I thought about applying the Floorplan method to it - but I am stumped. Coding isn't my natural environment and I am hoping somebody can help me.

![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/55891184/d9637885-3532-43d7-8ff2-e851acf61d48)

I have created this small SVG file to appear on my dashboard, that shows my 5 sensor values - and I want it to colour the tank on a gradient based off the sensor values roughly at each point. (eg blue for cold [<30°C], purple for warm [30><40°C], red for hot[<40°C]). I don't know if it's possible for 1 formula to do all that and create the gradient in one object, so I had thought about working around it by layering images on top of one another each representing a particular sensor like so

![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/55891184/1289676f-2b8e-4976-aabe-926a89290e25)

Given that I can't give the text sensors and their respective "part of the tank image" the same ID, the tank segments have just been named tank.1 etc. I've had no trouble in the past colouring my SVGs based off ON/OFF values, but this is way beyond me.

Is it possible to do this or am I barking up the wrong tree? Any help or a point in the right direction would be hugely appreciated!

Regards,

Craig

**Comment by OzGav** [comment]
Off the top of my head you could make five images and use `image_set` to show the one you want based on bands of temperature?

Also have a look a the home example which shows a moisture sensor graph with a gradient colour https://experiencelovelace.github.io/ha-floorplan/docs/example-home/

**Reply by r200cjt** [reply]
hey sorry for the delayed response! The graph gradient would work well - im just a little unsure how to actually impliment it....

I can create a template sensor which would tell me what % of hot water I have, but how would I go about converting that to a gradient fill in an object?

**Reply by OzGav** [reply]
You should be able to copy and paste the yaml from the example and change the entity to yours. This is a similar discussion here https://github.com/ExperienceLovelace/ha-floorplan/discussions/204  and more discussion here https://github.com/ExperienceLovelace/ha-floorplan/discussions/40

Look at the example SVG and you will see you need to add the image with a gradient to your floorplan. I haven't done this myself but have a Google for how to set the clip path. Here is one possible link https://design.tutsplus.com/tutorials/quick-tip-what-are-clipping-and-masking-in-inkscape--vector-24947

Additionally, if you aren't wedded to the idea of a vertical gauge you could use a semi circular one. See the same example for this
![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/19848947/facf43d5-9f29-4408-8374-f36d6a4b8a50)

**Reply by exetico** [reply]
So, how did it go? 

Did you end up with something like OzGavs suggestion, or, another solution? It's possible to do with CSS. 

Can't that kind of option be used on a SVG element?
https://developer.mozilla.org/en-US/docs/Web/CSS/gradient/linear-gradient

Else, it's possible to create the gradient in Inkscape, if I remember correcly. If that's true, you can find inspiration in the way Inkscape style the element, and use that somehow. I know it's a step up, but I'm just here throwing ideas, now that I can't find a follow up on how you ended up creating it :smile: .

**Reply by r200cjt** [reply]
Sadly I never got anywhere with it... I just couldn't figure it out - maybe I'm just not cut out for it 🤣 

Sadly the original idea was never one of my own, and is taken from the work of another.
https://github.com/davet2001/miscellaneous/blob/91311c7adada9e0491dfe8fd81ecd5d2bc43b551/dhw_tank.cgi

This method involves running that script on an Apache server and using CGI-Bin to generate the SVG. This works great - when I am at home. However I am one of these sad people that doesn't have a static IP, so I can't view it from outside home. I can create template sensors etc to show me the volume of hot water I have remaining - but lacking that visual just annoys me - that's when I thought about implementing the floorplan card. So while at home, it looks like this:

![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/55891184/ddea6bc3-9a31-4b72-aa97-8224bddb8d7f)

Would love to get it working, the linear gradient may work well - I'm not sure how I would go about setting points for each of my sensors? Or do I just convert the total volume of hot water to a percentage and let the gradient fill off that?

**Reply by OzGav** [reply]
You can get a free domain name here to solve your static ip problem https://www.noip.com

I would convert the volume to a percentage. You should be able to get that to work.

**Reply by r200cjt** [reply]
sadly my ISP uses a massive GNAT and trying to reroute traffic through NOIP or duckdns etc doesn't wanna work :'( I'll try to carve out a bit of time some evening this week and get another blast at getting this working :)
